lib/Swagger/src/SiteAttribute/Api/SiteAttributeApi.php

```diff
@@ -0,0 +1,535 @@
+            );
+        }
+        // default format to json
+        $resourcePath = str_replace("{format}", "json", $resourcePath);
+
+        
+        // for model (json/xml)
+        if (isset($_tempBody)) {
+            $httpBody = $_tempBody; // $_tempBody is the method argument, if present
+        } elseif (count($formParams) > 0) {
+            $httpBody = $formParams; // for HTTP post (form)
+        }
+        // this endpoint requires API key authentication
+        $apiKey = $this->apiClient->getApiKeyWithPrefix('X-Wikia-AccessToken');
+        if (strlen($apiKey) !== 0) {
+            $headerParams['X-Wikia-AccessToken'] = $apiKey;
+        }
+        // this endpoint requires API key authentication
+        $apiKey = $this->apiClient->getApiKeyWithPrefix('X-Wikia-UserId');
+        if (strlen($apiKey) !== 0) {
+            $headerParams['X-Wikia-UserId'] = $apiKey;
+        }
+        // make the API Call
+        try {
+            list($response, $statusCode, $httpHeader) = $this->apiClient->callApi(
+                $resourcePath,
+                'DELETE',
+                $queryParams,
+                $httpBody,
+                $headerParams,
+                null,
+                '/site/{siteId}/attr/{attrName}'
+            );
+
+            return array(null, $statusCode, $httpHeader);
+        } catch (ApiException $e) {
+            switch ($e->getCode()) {
+            }
+
+            throw $e;
+        }
+    }
+
+    /**
+     * Operation getAllAttributes
+     *
+     * Returns all available attributes for the specified siteId
+     *
+     * @param int $site_id The ID of the site (required)
+     * @return \Swagger\Client\SiteAttribute\Models\AllSiteAttributesHalResponse
+     * @throws \Swagger\Client\ApiException on non-2xx response
+     */
+    public function getAllAttributes($site_id)
+    {
+        list($response) = $this->getAllAttributesWithHttpInfo($site_id);
+        return $response;
+    }
+
+    /**
+     * Operation getAllAttributesWithHttpInfo
+     *
+     * Returns all available attributes for the specified siteId
+     *
+     * @param int $site_id The ID of the site (required)
+     * @return array of \Swagger\Client\SiteAttribute\Models\AllSiteAttributesHalResponse, HTTP status code, HTTP response headers (array of strings)
+     * @throws \Swagger\Client\ApiException on non-2xx response
+     */
+    public function getAllAttributesWithHttpInfo($site_id)
+    {
+        // verify the required parameter 'site_id' is set
+        if ($site_id === null) {
+            throw new \InvalidArgumentException('Missing the required parameter $site_id when calling getAllAttributes');
+        }
+        // parse inputs
+        $resourcePath = "/site/{siteId}/attr";
+        $httpBody = '';
+        $queryParams = array();
+        $headerParams = array();
+        $formParams = array();
+        $_header_accept = $this->apiClient->selectHeaderAccept(array('application/hal+json; charset=UTF-8'));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "application/hal+json; charset=UTF-8" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)